### ISSUE TITLE
shellcheck should exclude SC1091 by default since by default it will only warn about issues in the specified files

### ISSUE THREAD

**Issue by coiby** [koalaman/shellcheck#2539]
#### For bugs
- Rule Id (if any, e.g. SC1091): 
- My shellcheck version (`shellcheck --version` or "online"): 
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit
#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash
source abc
```

#### Here's what shellcheck currently says:
```
[Line 2:](javascript:setPosition(2, 8))
source abc
       ^-- [SC1091](https://www.shellcheck.net/wiki/SC1091) (info): Not following: abc was not specified as input (see shellcheck -x).
```

#### Here's what I wanted or expected to see:

No warning.

**Comment by koalaman** [comment]
ShellCheck actually requires the sourced file to give correct warnings about the *current* file. Consider something like:

```
#!/bin/bash
verbose=1
source abc
```

Should it trigger an unused assignment warning? It depends on whether `abc` uses it or not, and ShellCheck can only determine this by looking at it.
